liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/AbstractLibertySupport.java

```diff
@@ -112,4 +132,194 @@ public boolean include(Artifact artifact) {
+     *
+     *
+     * @param item  The item to create an artifact for; must not be null
+     * @return      The artifact for the given item
+     *
+     * @throws MojoExecutionException   Failed to create artifact
+     */
+    @Override
+    protected Artifact getArtifact(final ArtifactItem item) throws MojoExecutionException {
+        assert item != null;
+        
+        // resolve version in case it is a range
+        if (item.getVersion() != null) {
+            try {
+                resolveVersionRange(item);
+            } catch (VersionRangeResolutionException e) {
+                throw new MojoExecutionException(e.getLocalizedMessage(), e);
+            }
+        }
+        
+        // Return the artifact from the project dependency if it is available and the mojo 
+        // should have requiresDependencyResolution=ResolutionScope.COMPILE_PLUS_RUNTIME set 
+        Artifact artifact = resolveFromProjectDependencies(item);
+        
+        if (artifact != null) {
+            if (item.getVersion() == null) {
+                item.setVersion(artifact.getVersion());
+            }
+            // To maintain existing behavior, when the version is set in ArtifactItem which is
+            // different from dependencies configuration, the version in ArtifactItem takes 
+            // precedence.  
+            if (artifact.isResolved() && artifact.getVersion().equals(item.getVersion())) {
+                return artifact;
+            } else {
+                // create artifact from item
+                return createArtifact(item);
+            }
+        } else {
+            // if item has no version set, try to get it from the project dependencyManagement section
+            if (item.getVersion() == null && resolveFromProjectDepMgmt(item) != null) {
+                // get version from dependencyManagement
+                item.setVersion(resolveFromProjectDepMgmt(item).getVersion());
+            }
+            // create artifact from item
+            return createArtifact(item);
+        }
+    }
+
+    /**
+     * Create a new artifact.
+     *
+     * @param item  The item to create an artifact for
+     * @return      A resolved artifact for the given item.
+     *
+     * @throws MojoExecutionException   Failed to create artifact
+     */
+    @Override
+    protected Artifact createArtifact(final ArtifactItem item) throws MojoExecutionException {
+        assert item != null;
+        
+        if (item.getVersion() == null) {
+            throw new MojoExecutionException("Unable to find artifact version of " + item.getGroupId()
+                + ":" + item.getArtifactId() + " in either project dependencies or in project dependencyManagement.");
+        }
+        
+        // if version is a range get the highest available version
+        if (item.getVersion().trim().startsWith("[") || item.getVersion().trim().startsWith("(") ) {
```

Comment:
I am coming to grips with what's going on here.  Seems like we should not have to do all the manual work of figuring out the highest version.  Does it come down to `this.repositorySystem.resolveArtifact` doesn't resolve a range?  Seems like that should just be baked in.

Reply:
Unfortunately, `this.repositorySystem.resolveArtifact` doesn't resolve the range.  The constructor of `org.eclipse.aether.artifact.DefaultArtifact` doesn't take a versionRange and  we need to resolve it first.  I do find bug in the condition to detect the version range, that need to be fix.

Reply:
but if it is added as a dependency, it just works?

Reply:
Let me know when you want me to review the next commit. (Not that I caught the bug.)

Reply:
If it added as a dependency, the `project.getActifacts()` call will resolve the versionRange. 

Also I double check the versionRange detection condition, it is actually correct and no missing condition.  

However, I make a small change in  `AbstractLibertySupport.getArtifact()` method to make it more clear that if version is specified in ArtifactItem, it will override whatever is set in the project dependency and call createArtifact directly.